effekt/shared/src/main/scala/effekt/source/Tree.scala

```diff
@@ -541,52 +542,59 @@ export MatchPattern.*
 sealed trait Type extends Tree
 
 /**
- * Value Types
+ * Trees that represent inferred or synthesized *value* types (not present in the source = not parsed)
  */
-enum ValueType extends Type {
+case class ValueTypeTree(tpe: symbols.ValueType) extends Type
 
-  /**
-   * Trees that represent inferred or synthesized types (not present in the source)
-   */
-  case ValueTypeTree(tpe: symbols.ValueType)
+/**
+ * Trees that represent inferred or synthesized *block* types (not present in the source = not parsed)
+ */
+case class BlockTypeTree(eff: symbols.BlockType) extends Type
 
-  /**
-   * Types of first-class functions
-   */
-  case BoxedType(tpe: BlockType, capt: CaptureSet)
+/*
+ * Reference to a type, potentially with bound occurences in `args`
+ */
+case class TypeRef(id: IdRef, args: List[ValueType]) extends Type, Reference
 
-  // Bound occurrences (args can be empty)
-  case ValueTypeRef(id: IdRef, args: List[ValueType]) extends ValueType, Reference
-}
-export ValueType.*
+/**
+ * Types of first-class computations
+ */
+case class BoxedType(tpe: BlockType, capt: CaptureSet) extends Type
 
 /**
- * Block Types
+ * Types of (second-class) functions
  */
-enum BlockType extends Type {
+case class FunctionType(tparams: List[Id], vparams: List[ValueType], bparams: List[(Option[IdDef], BlockType)], result: ValueType, effects: Effects) extends Type
 
-  /**
-   * Trees that represent inferred or synthesized types (not present in the source)
-   */
-  case BlockTypeTree(eff: symbols.BlockType)
-  case FunctionType(tparams: List[Id], vparams: List[ValueType], bparams: List[(Option[IdDef], BlockType)], result: ValueType, effects: Effects)
-  case BlockTypeRef(id: IdRef, args: List[ValueType]) extends BlockType, Reference
-}
+/**
+ * Type-and-effect annotations
+ */
+case class Effectful(tpe: ValueType, eff: Effects) extends Type
+
+// These are just type aliases for documentation purposes.
+type BlockType = Type
+type ValueType = Type
```

Comment:
I've kept `BlockType` and `ValueType` around, since I think they're helpful for documentation purposes. Is that actually true?